### PR TITLE
drivers: nrf_qspi_nor: Avoid using QSPI with HFCLK192M divided by 4

### PR DESCRIPTION
**Pull request by anangl**
The nRF5340 product specification states that the QSPI peripheral supports 192 MHz and 96 MHz PCLK192M frequency, thus the default /4 divider for HFCLK192M (from which PCLK192M is derived) can only be used when the QSPI peripheral is idle. Correct the shim accordingly and improve related comments.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

### CODE REVIEW

File: drivers/flash/nrf_qspi_nor.c [hunk 1]
```diff
@@ -673,9 +673,24 @@ static int qspi_nrfx_configure(const struct device *dev)
 	struct qspi_nor_data *dev_data = dev->data;
 	const struct qspi_nor_config *dev_config = dev->config;
 
+#if defined(BASE_CLOCK_DIV)
```
Comment (de-nordic) [inline drivers/flash/nrf_qspi_nor.c:676]: I know that the `BASE_CLOCK_DIV` is currently only defined when `CONFIG_SOC_SERIES_NRF53X` is defined, but isn't it better to use `CONFIG_SOC_SERIES_NRF53X` here to notify that this code will be included only for the NRF53 series?

File: drivers/flash/nrf_qspi_nor.c [hunk 2]
```diff
@@ -673,9 +673,24 @@ static int qspi_nrfx_configure(const struct device *dev)
...
+	 * divider.
+	 */
+	nrf_clock_hfclk192m_div_set(NRF_CLOCK, BASE_CLOCK_DIV);
+#endif
+
 	nrfx_err_t res = nrfx_qspi_init(&dev_config->nrfx_cfg,
 					qspi_handler,
 					dev_data);
+
+#if defined(BASE_CLOCK_DIV)
```
Comment (de-nordic) [inline drivers/flash/nrf_qspi_nor.c:689]: Same question as above.

File: drivers/flash/nrf_qspi_nor.c (hunk 1 above)
Comment (anangl) [reply drivers/flash/nrf_qspi_nor.c:676]: Yes, this will be clearer. I'll update all occurrences of `#if defined(BASE_CLOCK_DIV)`.

Comment (anangl) [reply drivers/flash/nrf_qspi_nor.c]: Done.

**Review by de-nordic — APPROVED** [review]
Looks OK. Thanks!

**Review by stephanosio — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**